Ladybird/Headless/Test.cpp

```diff
@@ -29,6 +29,63 @@
 
 namespace Ladybird {
 
+static AK::Vector<AK::StringView> current_os_list()
+{
+    // This is a silly name.
+    AK::Vector<AK::StringView> oses = {};
+
+#ifdef AK_OS_ANDROID
+    oses.append("android"sv);
+#endif
+#ifdef AK_OS_BSD_GENERIC
+    oses.append("bsd_generic"sv);
+#endif
+#ifdef AK_OS_DRAGONFLY
+    oses.append("dragonfly"sv);
+#endif
+#ifdef AK_OS_EMSCRIPTEN
+    oses.append("emscripten"sv);
+#endif
+#ifdef AK_OS_FREEBSD
+    oses.append("freebsd"sv);
+#endif
+#ifdef AK_OS_GNU_HURD
+    oses.append("gnu_hurd"sv);
+#endif
+#ifdef AK_OS_HAIKU
+    oses.append("haiku"sv);
+#endif
+#ifdef AK_OS_IOS
+    oses.append("ios"sv);
+#endif
+#ifdef AK_OS_LINUX
+    oses.append("linux"sv);
+#endif
+#ifdef AK_OS_MACH
+    oses.append("mach"sv);
+#endif
+#ifdef AK_OS_MACOS
```

Comment:
I'm not sure we really *need* to enumerate every possible OS flag here. I'd just start with something simple, since in practice, we only really need to differentiate between Linux vs macOS (and eventually Windows).

```c++
static constexpr StringView current_os()
{
#if defined(AK_OS_LINUX)
    return "Linux"sv;
#elif defined(AK_OS_MACOS)
    return "macOS"sv;
#elif defined AK_OS_WINDOWS
    return "Windows"sv;
#else
    return "Other"sv; // or an empty string, or something else
#endif
}
```

Reply:
I mostly did it because it was easy (just copy-pasted all the `#defines` from `AK/Platform.h` and did some light multi-cursor formatting) and I wasn't sure what platforms/flags would end up being useful. That being said though, we can always add/tweak the set as needed later.